Userland/Libraries/LibJS/Bytecode/Instruction.h

```diff
@@ -95,6 +95,9 @@ class Instruction {
     void replace_references(BasicBlock const&, BasicBlock const&);
     static void destroy(Instruction&);
 
+    Vector<Register*> write_registers();
```

Comment:
You can use `Vector<Register&>` here, it will work correctly and avoid a bunch of manual `&`'s and `*`'s

Reply:
Vectors of references are much more of a pain to deal with. you can initialize then with an `initializer_list`, and append didn't work for me either.

Reply:
append should just work with `T&`, if not...it's a bug
`initializer_list`...I'm not sure how to deal with, do you have many of those?

Reply:
to the append
you would need to move into the vector, to not accidentally clean up the data stored in there

Reply:
most `{read, write}_register_impl()` just return an `initializer_list`

Reply:
I think we should switch that Vector ctor to `Vector(Ts&&...)`, taking an initialized_list isn't nice anyway.
If that works out, I'll push it to master and you'll get your free nice returns.
But please do give me an MVE of that append() issue.

Reply:
Update on that `Vector(Ts&&...)`, while we can _add_ it, we can't quite replace the initializer list ctor with it.
so I guess this bit is resolved in a sense.